packages/core/src/render3/definition.ts

```diff
@@ -696,6 +696,10 @@ function getComponentId(componentDef: ComponentDef<unknown>): string {
     componentDef.decls,
     componentDef.encapsulation,
     componentDef.standalone,
+    componentDef.signals,
+    componentDef.exportAs,
+    componentDef.inputs,
+    componentDef.outputs,
```

Comment:
Since these are objects the .toString will return `[object Object]`

```suggestion
    Object.entries(componentDef.inputs)
    Object.entries(componentDef.outputs),
```

Reply:
Good catch! I've wrapped them into `JSON.stringify` :)